modules/system/models/MailTemplate.php

```diff
@@ -68,7 +69,12 @@ public static function allTemplates()
         $codes = array_keys(self::listAllTemplates());
 
         foreach ($codes as $code) {
-            $result[] = self::findOrMakeTemplate($code);
+            try {
+                $result[] = self::findOrMakeTemplate($code);
+            }
+            catch (InvalidArgumentException $e) {
```

Comment:
Does view::exists() throw this exception?

Reply:
Also self::whereCode() @LukeTowers

Reply:
> Also `self::whereCode()` @LukeTowers

Reply:
Actually, `View::exists($code)` does NOT throw the exception.

Reply:
So do we need this try catch block?

Reply:
Yes we do, that's how we can skip the templates that have been orphaned.

Reply:
What throws that exception?

Reply:
@LukeTowers If you mean the origin of the throw (vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php line 112):
```
if (! isset($this->hints[$segments[0]])) {
   throw new InvalidArgumentException("No hint path defined for [{$segments[0]}].");
}
```

If you mean the call within MailTemplate that leads to the throw, that would be `self::whereCode()` in:
```
    public static function findOrMakeTemplate($code)
    {   
        $template = self::whereCode($code)->first();

        if (!$template && View::exists($code)) {
            $template = new self;
            $template->code = $code;
            $template->fillFromView($code);
        }

        return $template;
    }
```

Reply:
I couldn't find the definition of the `whereCode()` method anywhere... any idea?

Reply:
It's little known shorthand for `where('code', $value)` implemented in the Laravel Query builder. What's the stack trace of the exception?

Reply:
```
[2020-10-20 16:10:27] dev.ERROR: InvalidArgumentException: No hint path defined for [offline.mall]. in /home/r4l/public_html/vendor/laravel/framework/src/Illu
minate/View/FileViewFinder.php:112
Stack trace:
#0 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php(90): Illuminate\View\FileViewFinder->parseNamespaceSegments('offline.
mall::m...')
#1 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/View/FileViewFinder.php(76): Illuminate\View\FileViewFinder->findNamespacedView('offline.mall
::m...')
#2 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/View/Factory.php(131): Illuminate\View\FileViewFinder->find('offline.mall::m...')
#3 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\View\Factory->make('offline.mall::m...')
#4 /home/r4l/public_html/modules/system/models/MailTemplate.php(155): Illuminate\Support\Facades\Facade::__callStatic('make', Array)
#5 /home/r4l/public_html/modules/system/models/MailTemplate.php(140): System\Models\MailTemplate::getTemplateSections('offline.mall::m...')
#6 /home/r4l/public_html/modules/system/models/MailTemplate.php(129): System\Models\MailTemplate->fillFromView('offline.mall::m...')
#7 /home/r4l/public_html/vendor/october/rain/src/Database/Model.php(168): System\Models\MailTemplate->afterFetch()
#8 /home/r4l/public_html/vendor/october/rain/src/Events/Dispatcher.php(417): October\Rain\Database\Model->October\Rain\Database\{closure}(Object(System\Models
\MailTemplate))
#9 /home/r4l/public_html/vendor/october/rain/src/Events/Dispatcher.php(260): October\Rain\Events\Dispatcher->October\Rain\Events\{closure}('eloquent.fetche...
', Array)
#10 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Concerns/HasEvents.php(189): October\Rain\Events\Dispatcher->dispatch('elo
quent.fetche...', Array)
#11 /home/r4l/public_html/vendor/october/rain/src/Database/Model.php(444): Illuminate\Database\Eloquent\Model->fireModelEvent('fetched', false)
#12 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(322): October\Rain\Database\Model->newFromBuilder(Object(stdCl
ass))
#13 [internal function]: Illuminate\Database\Eloquent\Builder->Illuminate\Database\Eloquent\{closure}(Object(stdClass))
#14 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(323): array_map(Object(Closure), Array)
#15 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Support/Traits/ForwardsCalls.php(23): Illuminate\Database\Eloquent\Builder->hydrate(Array)
#16 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Model.php(1622): Illuminate\Database\Eloquent\Model->forwardCallTo(Object(October\Rain\Database\Builder), 'hydrate', Array)
#17 /home/r4l/public_html/vendor/october/rain/src/Extension/ExtendableTrait.php(428): Illuminate\Database\Eloquent\Model->__call('hydrate', Array)
#18 /home/r4l/public_html/vendor/october/rain/src/Database/Model.php(650): October\Rain\Database\Model->extendableCall('hydrate', Array)
#19 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(545): October\Rain\Database\Model->__call('hydrate', Array)
#20 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(529): Illuminate\Database\Eloquent\Builder->getModels(Array)
#21 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(143): Illuminate\Database\Eloquent\Builder->get(Array)
#22 /home/r4l/public_html/modules/system/models/MailTemplate.php(160): Illuminate\Database\Eloquent\Builder->first()
#23 /home/r4l/public_html/modules/system/models/MailTemplate.php(77): System\Models\MailTemplate::findOrMakeTemplate('offline.mall::m...')
#24 /home/r4l/public_html/plugins/rainlab/translate/classes/ThemeScanner.php(125): System\Models\MailTemplate::allTemplates()
#25 /home/r4l/public_html/plugins/rainlab/translate/classes/ThemeScanner.php(56): RainLab\Translate\Classes\ThemeScanner->scanMailTemplatesForMessages()
#26 /home/r4l/public_html/plugins/rainlab/translate/classes/ThemeScanner.php(28): RainLab\Translate\Classes\ThemeScanner->scanForMessages()
#27 /home/r4l/public_html/plugins/rainlab/translate/controllers/Messages.php(76): RainLab\Translate\Classes\ThemeScanner::scan()
#28 [internal function]: RainLab\Translate\Controllers\Messages->onScanMessages()
#29 /home/r4l/public_html/modules/backend/classes/Controller.php(628): call_user_func_array(Array, Array)
#30 /home/r4l/public_html/modules/backend/classes/Controller.php(478): Backend\Classes\Controller->runAjaxHandler('onScanMessages')
#31 /home/r4l/public_html/modules/backend/classes/Controller.php(277): Backend\Classes\Controller->execAjaxHandlers()
#32 /home/r4l/public_html/modules/backend/classes/BackendController.php(165): Backend\Classes\Controller->run('index', Array)
#33 [internal function]: Backend\Classes\BackendController->run('rainlab/transla...')
#34 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array(Array, Array)
#35 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\Routing\Controller->callAction('run', Array)
#36 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(219): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Backend\Classes\BackendController), 'run')
#37 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Route.php(176): Illuminate\Routing\Route->runController()
#38 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(681): Illuminate\Routing\Route->run()
#39 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#40 /home/r4l/public_html/modules/backend/classes/BackendController.php(68): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#41 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(153): Backend\Classes\BackendController->Backend\Classes\{closure}(Object(Illuminate\Http\Request), Object(Closure))
#42 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(41): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#43 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Routing\Middleware\SubstituteBindings->handle(Object(Illuminate\Http\Request), Object(Closure))
#44 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#45 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure)) 
#46 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(56): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#47 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#48 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#49 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(:
Object(Illuminate\Http\Request), Object(Closure))
#50 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#53 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(683): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#54 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(658): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#55 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Routing/Router.php(624): Illuminate\Routing\Router->runRoute(Object(Illuminate\Http\Request), Object(Illuminate\Routing\Route))
#56 /home/r4l/public_html/vendor/october/rain/src/Router/CoreRouter.php(20): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request)) 
#57 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(170): October\Rain\Router\CoreRouter->dispatch(Object(Illuminate\Http\Request)) 
#58 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(130): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#59 /home/r4l/public_html/plugins/rainlab/debugbar/middleware/InjectDebugbar.php(30): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) 
#60 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): RainLab\Debugbar\Middleware\InjectDebugbar->handle(Object(Illuminate\Http\Request), Object(Closure)) 
#61 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(63): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) 
#62 /home/r4l/public_html/vendor/october/rain/src/Foundation/Http/Middleware/CheckForMaintenanceMode.php(25): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure)) 
#63 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(171): October\Rain\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure)) 
#64 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(105): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request)) 
#65 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(145): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#66 /home/r4l/public_html/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#67 /home/r4l/public_html/index.php(43): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request)) 
#68 {main}  
```

Reply:
Looks like what's actually causing the exception is the fillFromView() call in `MailTemplate::afterFetch()`

So a try/catch around this call would do the trick.

Reply:
And ultimately, the call that causes this in View::make() in getTemplateSections()... so this is another fix:
```
    protected static function getTemplateSections($code)
    {   
        try { 
            $view = View::make($code);
            return MailParser::parse(FileHelper::get($view->getPath()));
        }
        catch (\Exception $e) {
            return null;
        }
    }       
```